### ISSUE TITLE
Flannel v0.13.0 segfaults on Fedora 33 (fine on Fedora 32)

### ISSUE THREAD

**Issue by AkihiroSuda** [flannel-io/flannel#1365]
## Expected Behavior
Flannel should not segfault

## Current Behavior
Flannel segfaults on Fedora 33. Works fine on Fedora 32.

## Steps to Reproduce (for bugs)
Just running `flanneld -version` is enough to trigger the issue.

Fedora 33 (segfaults)
```console
$ docker run -it --rm fedora:33
[root@24377f40f9e8 /]# curl -sSL https://github.com/coreos/flannel/releases/download/v0.13.0/flannel-v0.13.0-linux-amd64.tar.gz | tar Cxzv /usr/local/bin flanneld
flanneld
[root@24377f40f9e8 /]# flanneld -version
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0xe5 pc=0x7fe17834bc64]

runtime stack:
runtime.throw(0x1a32f41, 0x2a)
        /usr/local/go/src/runtime/panic.go:1116 +0x72
runtime.sigpanic()
        /usr/local/go/src/runtime/signal_unix.go:679 +0x46a

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x151cfe0, 0xc000067c30, 0xc000010030)
        /usr/local/go/src/runtime/cgocall.go:133 +0x5b fp=0xc000067c00 sp=0xc000067bc8 pc=0x404fdb
os/user._Cfunc_mygetpwuid_r(0x0, 0xc000094c90, 0x45e1200, 0x400, 0xc000010030, 0xc000000000)
        _cgo_gotypes.go:173 +0x4d fp=0xc000067c30 sp=0xc000067c00 pc=0x733d4d
os/user.lookupUnixUid.func1.1(0x0, 0xc000094c90, 0xc00006ac40, 0xc000010030, 0xc000067cd0)
        /usr/local/go/src/os/user/cgo_lookup_unix.go:103 +0xe4 fp=0xc000067c80 sp=0xc000067c30 pc=0x7351f4
os/user.lookupUnixUid.func1(0x1838240)
        /usr/local/go/src/os/user/cgo_lookup_unix.go:103 +0x45 fp=0xc000067cb8 sp=0xc000067c80 pc=0x735255
os/user.retryWithBuffer(0xc00006ac40, 0xc000067d90, 0x0, 0x0)
        /usr/local/go/src/os/user/cgo_lookup_unix.go:247 +0x3b fp=0xc000067d10 sp=0xc000067cb8 pc=0x7347fb
os/user.lookupUnixUid(0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/os/user/cgo_lookup_unix.go:96 +0x12d fp=0xc000067dd8 sp=0xc000067d10 pc=0x73416d
os/user.current(0x0, 0x0, 0x0)
        /usr/local/go/src/os/user/cgo_lookup_unix.go:49 +0x49 fp=0xc000067e18 sp=0xc000067dd8 pc=0x734009
os/user.Current.func1()
        /usr/local/go/src/os/user/lookup.go:15 +0x22 fp=0xc000067e40 sp=0xc000067e18 pc=0x7350a2
sync.(*Once).doSlow(0x28cc5e0, 0x1a8b500)
        /usr/local/go/src/sync/once.go:66 +0xec fp=0xc000067e90 sp=0xc000067e40 pc=0x47e13c
sync.(*Once).Do(...)
        /usr/local/go/src/sync/once.go:57
os/user.Current(0xc0000402f4, 0xc, 0x1a05595)
        /usr/local/go/src/os/user/lookup.go:15 +0xf2 fp=0xc000067ec0 sp=0xc000067e90 pc=0x733a72
github.com/golang/glog.init.1()
        /go/src/github.com/coreos/flannel/vendor/github.com/golang/glog/glog_file.go:63 +0x41 fp=0xc000067f28 sp=0xc000067ec0 pc=0x739721
runtime.doInit(0x2862320)
        /usr/local/go/src/runtime/proc.go:5420 +0x8a fp=0xc000067f58 sp=0xc000067f28 pc=0x4441fa
runtime.doInit(0x28673c0)
        /usr/local/go/src/runtime/proc.go:5415 +0x57 fp=0xc000067f88 sp=0xc000067f58 pc=0x4441c7
runtime.main()
        /usr/local/go/src/runtime/proc.go:190 +0x1b6 fp=0xc000067fe0 sp=0xc000067f88 pc=0x437666
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc000067fe8 sp=0xc000067fe0 pc=0x4650d1

goroutine 6 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x28cd060)
        /go/src/github.com/coreos/flannel/vendor/github.com/golang/glog/glog.go:879 +0x8b
created by github.com/golang/glog.init.0
        /go/src/github.com/coreos/flannel/vendor/github.com/golang/glog/glog.go:410 +0x26f
```

Fedora 32 (works fine)

```console
$ docker run -it --rm fedora:32
[root@b9d08d5bca88 /]# curl -sSL https://github.com/coreos/flannel/releases/download/v0.13.0/flannel-v0.13.0-linux-amd64.tar.gz | tar Cxzv /usr/local/bin flanneld
flanneld
[root@b9d08d5bca88 /]# flanneld -version
v0.13.0
[root@b9d08d5bca88 /]# 
```

## Your Environment

* Flannel version: v0.13.0
* Operating System and version: Fedora 32 & 33

**Comment by luthermonson** [comment]
@AkihiroSuda seems like this is deep in go and we tried building with go1.15.2 and looks like this was fixed. We will try a version bump for building flanneld from go1.14.6 to go1.15.2 and see if this solves this issue.

**Comment by donbowman** [comment]
same behaviour observed on Ubuntu 20.10, same stack trace.

**Comment by luthermonson** [comment]
going to 1.15 did not fix this, it's a cgo call to getpwuid that fails, i dont know how we can fix this without go addressing the problem first.

**Comment by donbowman** [comment]
it did fix it for me on ubuntu 20.10 w/ golang 1.15.6 My stack trace does/did match above.
I think(?) its related to the changes in nsswitch.conf, but am not sure.

ah! i se, it fixed for me since i linked dynamic. If you static link, it fails, if you dyamic link it fails.
